src/core/layout/qgslayoutitemchart.cpp

```diff
@@ -0,0 +1,390 @@
+/***************************************************************************
+                         qgslayoutitemchart.cpp
+                         -------------------
+     begin                : August 2025
+     copyright            : (C) 2025 by Mathieu
+     email                : mathieu at opengis dot ch
+***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslayoutitemchart.h"
+#include "moc_qgslayoutitemchart.cpp"
+#include "qgsapplication.h"
+#include "qgslayoutitemregistry.h"
+#include "qgslayout.h"
+#include "qgslayoutrendercontext.h"
+#include "qgslayoutreportcontext.h"
+#include "qgslayoututils.h"
+#include "qgsplotregistry.h"
+
+#include <QDomDocument>
+#include <QDomElement>
+#include <QPainter>
+
+QgsLayoutItemChart::QgsLayoutItemChart( QgsLayout *layout )
+  : QgsLayoutItem( layout )
+{
+  // default to no background
+  setBackgroundEnabled( false );
+
+  mPlot.reset( dynamic_cast<Qgs2DPlot *>( QgsApplication::instance()->plotRegistry()->createPlot( "line" ) ) );
+
+  mGathererTimer.setInterval( 10 );
+  mGathererTimer.setSingleShot( true );
+  connect( &mGathererTimer, &QTimer::timeout, this, &QgsLayoutItemChart::gatherData );
+}
+
+int QgsLayoutItemChart::type() const
+{
+  return QgsLayoutItemRegistry::LayoutChart;
+}
+
+QIcon QgsLayoutItemChart::icon() const
+{
+  return QgsApplication::getThemeIcon( QStringLiteral( "/mLayoutItemChart.svg" ) );
+}
+
+QgsLayoutItemChart *QgsLayoutItemChart::create( QgsLayout *layout )
+{
+  return new QgsLayoutItemChart( layout );
+}
+
+void QgsLayoutItemChart::setPlot( QgsPlot *plot )
+{
+  Qgs2DPlot *plot2d = dynamic_cast<Qgs2DPlot *>( plot );
+  if ( !plot2d )
+  {
+    delete plot;
+    return;
+  }
+
+  // Logic to minimise plot data refresh to bare minimum
+  bool requireRefresh = !mPlot || !plot;
+  if ( mPlot && plot )
+  {
+    if ( mPlot->type() != plot->type() )
+    {
+      requireRefresh = true;
+    }
+    else
+    {
+      Qgs2DXyPlot *oldPlot2dXy = dynamic_cast<Qgs2DXyPlot *>( mPlot.get() );
+      Qgs2DXyPlot *newPlot2dXy = dynamic_cast<Qgs2DXyPlot *>( plot2d );
+      if ( oldPlot2dXy && oldPlot2dXy->xAxis().type() != newPlot2dXy->xAxis().type() )
+      {
+        requireRefresh = true;
+      }
```

Comment:
```suggestion
      if ( oldPlot2dXy && newPlot2dXy && oldPlot2dXy->xAxis().type() == newPlot2dXy->xAxis().type() )
      {
        // a special case, we don't need to refresh the plot data.
        requireRefresh = false;
      }
      else
      {
        requireRefresh = true;
      }
```

Reply:
Changed logic, but removed the "special" case as it's not that special :)